hub_updater.py

```diff
@@ -109,16 +109,22 @@ def create_pr(manifest_path, requirements_path, module, jina_core_version, hub_r
         pr_name = f'chore: testing/building {module} ({module_version}) on new jina core: {version["major"]}.{version["minor"]}.'
     pr: PullRequest = get_pr_from_gh(pr_name, all_prs)
     if pr:
-        print(
-            f'Warning: module {module} has already been tested on {module_version} with jina {jina_core_version}. '
-            f'Skipping...')
-        if pr.state == 'open':
-            # something must've stopped us from closing it
-            # make sure we close it
-            print(f'PR found open for {module} on Jina core v{jina_core_version}. Will handle now...')
-            return pr
+        if FORCE_RECHECK_PR:
+            print(f'Found existing PR for version: {pr.html_url}. Will rename as [old] and try again...')
+            pr.edit(
+                title=f'[old] {pr.title}'
+            )
         else:
-            return None
+            print(
+                f'Warning: module {module} has already been tested on {module_version} with jina {jina_core_version}. '
+                f'Skipping...')
+            if pr.state == 'open':
+                # something must've stopped us from closing it
+                # make sure we close it
+                print(f'PR found open for {module} on Jina core v{jina_core_version}. Will handle now...')
+                return pr
+            else:
+                return None
```

Comment:
Personally, I'm not a big fan of complex if-else. This is error-prone and hard to maintain.
```
if pr:
    if not FORCE_RECHECK_PR:
        print(
            f'Warning: module {module} has already been tested on {module_version} with jina {jina_core_version}. '
            f'Skipping...')
        if pr.state != 'open':
            return None
        # something must've stopped us from closing it
        # make sure we close it
        print(f'PR found open for {module} on Jina core v{jina_core_version}. Will handle now...')
        return pr
    print(f'Found existing PR for version: {pr.html_url}. Will rename as [old] and try again...')
    pr.edit(title=f'[old] {pr.title}')
```

Reply:
I refactored parts of the code here though. In refactoring, I focused on how positive conditionals are easier to read (`if FORCE_RECHECK` instead of `if not ...`)